### DISCUSSION TITLE
[Build Script][Linux & macOS]

### DISCUSSION THREAD

**Discussion by sn4k3** [emgucv/emgucv#718] [Show and tell]
I've created an script to facilitate the build of **libcvextern.so** working on the host system but also installs dependencies for them.

- Debian and variants
- Arch and variants
- RHEL and variants
- macOS x64 and arm64

Script: https://github.com/sn4k3/UVtools/blob/master/build/libcvextern.sh

- Place script on the target machine
- Execute
   - Installs the build dependencies and compile: `./libcvextern.sh -i`
   - Compile: `./libcvextern.sh`
   - Compile a tag: `./libcvextern.sh 4.5.5`
   - Clean: `./libcvextern.sh clean`

**Comment by mexico-johnny** [comment]
Nicely done!

Any chance you’d be interested in expanding platforms to support NixOS x64/22.05 (https://nixos.org/)?

I’ve been trying to get a custom build working for the last several days, but so far - no luck.

**Reply by sn4k3** [reply]
This script compile no custom version, it use the mini preset. 
By my experience you need to compile this on ubuntu, any other cause error to me. My ubuntu build works in all other systems

**Reply by jd-talos** [reply]
Thx.  I've been able to build successfully on NixOS 22.11 without issues following the instructions outlined [here](https://www.emgu.com/wiki/index.php/Download_And_Installation#Linux), however, execution of 'BuildInfo.NetCore.Console]' fails with 'System.DllNotFoundException: Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcvextern: cannot open shared object file: No such file or directory ...'

Executing 'LG_DEBUG=libs dotnet run' results in the following errors:

`
     27159:	dotnet: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27159:	/nix/store/4axlc21msjr9g181l2y6hxsmm4jar8i7-dotnet-sdk-6.0.403/shared/Microsoft.NETCore.App/6.0.11/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_RegisterModule (fatal)
     27159:	/nix/store/4axlc21msjr9g181l2y6hxsmm4jar8i7-dotnet-sdk-6.0.403/shared/Microsoft.NETCore.App/6.0.11/libclrjit.so: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27159:	dotnet: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27159:	/nix/store/4mxnw95jcm5a27qk60z7yc0gvxp42b9a-openssl-3.0.7/lib/libssl.so: error: symbol lookup error: undefined symbol: SSL_state (fatal)
     27159:	/nix/store/4mxnw95jcm5a27qk60z7yc0gvxp42b9a-openssl-3.0.7/lib/libssl.so: error: symbol lookup error: undefined symbol: ERR_put_error (fatal)
     27185:	/home/talos/projects/emgucv/Emgu.CV.Example/BuildInfo.NetCore.Console/bin/Debug/net6.0/BuildInfo.NetCore.Console: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27185:	/nix/store/4axlc21msjr9g181l2y6hxsmm4jar8i7-dotnet-sdk-6.0.403/shared/Microsoft.NETCore.App/6.0.11/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_RegisterModule (fatal)
     27185:	/nix/store/4axlc21msjr9g181l2y6hxsmm4jar8i7-dotnet-sdk-6.0.403/shared/Microsoft.NETCore.App/6.0.11/libclrjit.so: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27185:	/home/talos/projects/emgucv/Emgu.CV.Example/BuildInfo.NetCore.Console/bin/Debug/net6.0/BuildInfo.NetCore.Console: error: symbol lookup error: undefined symbol: DllMain (fatal)
     27185:	/home/talos/projects/emgucv/Emgu.CV.Example/BuildInfo.NetCore.Console/bin/Debug/net6.0/libcvextern.so: error: symbol lookup error: undefined symbol: png_get_valid (fatal)
     27185:	/home/talos/projects/emgucv/Emgu.CV.Example/BuildInfo.NetCore.Console/bin/Debug/net6.0/libcvextern.so: error: symbol lookup error: undefined symbol: png_get_valid (fatal)
   at Emgu.CV.CvInvoke.RedirectError(CvErrorCallback errorHandler, IntPtr userdata, IntPtr prevUserdata)
`

Thoughts?
